source/server/listener_manager_impl.cc

```diff
@@ -401,7 +401,7 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
   // Tell all workers to stop accepting new connections on this listener.
   draining_it->listener_->debugLog("draining listener");
   for (const auto& worker : workers_) {
-    worker->stopListener(*draining_it->listener_);
+    worker->stopListener(*draining_it->listener_, nullptr);
```

Comment:
Don't we want to close the accept socket in this case also? Shouldn't we share common code for this?

Reply:
Yeah. I think that makes sense. Probably we should refactor this to a method `stopListener` that takes worker and ListenerPtr?

Reply:
Only thing, I am not sure is if we want to increment the `stopped_counter` stat here in this case?